code/Model/SiteTree.php

```diff
@@ -3434,9 +3434,22 @@ protected function getExcludedURLSegments()
      */
     public function getAnchorsOnPage()
     {
+        $content = $this->Content ?? '';
+        
+        // Shortcodes may contain name/id attributes, they mess up the parsing below
+        // we replace them with their content first
+        // Note: We don't use ShortcodeParser::parse() here as that would execute the shortcodes
```

Comment:
What if the executing the shortcodes can bring in additional content to the page that would have anchors in it and from the perspective of the final rendered page one could expect anchors extracted for that as well?